packages/block-library/src/navigation-link/shared/controls.js

```diff
@@ -70,56 +70,75 @@ function getEntityTypeName( type, kind ) {
-
-	// Local state to control the input value
-	const [ inputValue, setInputValue ] = useState( url );
 
-	// Sync local state when url prop changes (e.g., from undo/redo or external updates)
-	useEffect( () => {
-		setInputValue( url );
-		lastURLRef.current = url;
-	}, [ url ] );
-
-	// Use the entity binding hook internally
-	const { hasUrlBinding, isBoundEntityAvailable, clearBinding } =
+	// Use the entity binding hook for UI state (help text, link preview, etc.)
+	const { hasUrlBinding, isBoundEntityAvailable, entityRecord } =
 		useEntityBinding( {
 			clientId,
 			attributes,
 		} );
 
-	// Get direct store dispatch to bypass setBoundAttributes wrapper
-	const { updateBlockAttributes } = useDispatch( blockEditorStore );
+	const needsHelpText = hasUrlBinding;
+	const helpText = isBoundEntityAvailable
+		? BindingHelpText( {
+				type: attributes.type,
+				kind: attributes.kind,
+		  } )
+		: MissingEntityHelpText( {
+				type: attributes.type,
+				kind: attributes.kind,
+		  } );
 
-	const unsyncBoundLink = () => {
-		// Clear the binding first
-		clearBinding();
+	// Get the link change handler with built-in binding management
+	const handleLinkChange = useHandleLinkChange( {
+		clientId,
+		attributes,
+		setAttributes,
+	} );
 
-		// Use direct store dispatch to bypass block bindings safeguards
-		// which prevent updates to bound attributes when calling setAttributes.
-		// setAttributes is actually setBoundAttributes, a wrapper function that
-		// processes attributes through the binding system.
-		// See: packages/block-editor/src/components/block-edit/edit.js
-		updateBlockAttributes( clientId, {
-			url: lastURLRef.current, // set the lastURLRef as the new editable value so we avoid bugs from empty link states
-			id: undefined,
-		} );
-	};
+	const linkTitle =
+		entityRecord?.title?.rendered ||
+		entityRecord?.title ||
+		entityRecord?.name;
 
-	useEffect( () => {
-		// Only want to focus the input if the url is not bound to an entity.
-		if ( ! hasUrlBinding && shouldFocusURLInputRef.current ) {
-			// focuses and highlights the url input value, giving the user
-			// the ability to delete the value quickly or edit it.
-			urlInputRef.current?.select();
-		}
-		shouldFocusURLInputRef.current = false;
-	}, [ hasUrlBinding ] );
+	const linkImage = useSelect(
+		( select ) => {
+			// Only fetch for post-type entities with featured media
+			if ( ! entityRecord || ! entityRecord.featured_media ) {
+				return null;
+			}
+
+			const { getEntityRecord } = select( coreStore );
+
+			// Get the media entity to fetch the image URL
+			const media = getEntityRecord(
+				'postType',
+				'attachment',
+				entityRecord.featured_media
+			);
+
+			// Return the thumbnail or medium size URL, fallback to source_url
+			return (
+				media?.media_details?.sizes?.thumbnail?.source_url ||
+				media?.media_details?.sizes?.medium?.source_url ||
+				media?.source_url ||
+				null
+			);
+		},
+		[ entityRecord ]
```

Comment:
If we're only consuming a specific prop of this `entityRecord` we could just have that as a dep to avoid this selector re-running right?